### PR TITLE
IGNITE-14441 hide sensitive data on log messages for long running que…

### PR DESCRIPTION
**Pull request by tledkov-gridgain**
…ry and huge results query according with the property IGNITE_TO_STRING_INCLUDE_SENSITIVE

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SensitiveInformationForLongRunningQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
```
Comment (nizhikov) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SensitiveInformationForLongRunningQueryTest.java:2]: Please, fix copyright.

**Comment by nizhikov** [comment]
Hello, @tledkov-gridgain 

Can you, please, clarify, why you treat the query plan as sensitive info?

Also, please, take a look at the #8821 
Is this patch you are looking for?

I think we should hide only actual values of the query parameters in hiding sensitive mode.
The query plan itself is not sensitive info.

So I suggest reusing logic implemented in `IgniteH2Indexing#sqlWithoutConst` in `H2QueryInfo#printLogMessage`.

**Comment by tledkov-gridgain** [comment]
> Can you, please, clarify, why you treat the query plan as sensitive info?
Please take a look on the usage of the `S#includeSensitive`. The class names and field names are hidden in log messages.
SQL queries and plans also contains names of the tables & fields and must be hidden in log messages.

I guess the behavior of hiding sensitive info must be consistent.

I don't see any reason to hide SQL constants at the system views because user may use parameters for sensitive info and access to system views must check security  permission.

**Comment by nizhikov** [comment]
>  The class names and field names are hidden in log messages.

But why? Do we have some definition of "sensitive" information?
I don't think class names and field names is sensitive information.

> I don't see any reason to hide SQL constants at the system views because user may use parameters for sensitive info and access to system views must check security permission.

Yes, you are correct.

1. Access to system views should be restricted. Please, take a look at IGNITE-14207
2. User may (but may not) use parameters. And we should help the user in both cases isn't it?

**Comment by tledkov-gridgain** [comment]
> But why? Do we have some definition of "sensitive" information?
> I don't think class names and field names is sensitive information.
I have no hard view about this. Bot now it is. I guess we start discussion on devlist to clarify the scope of the sensitive info and specify behavior of the Ignite's modules/subsystem (I mean logs, view etc).

**Comment by nizhikov** [comment]
> I guess we have to start discussion on devlist to clarify the scope of the sensitive info

Yes, let's do it.

**Comment by ygerzhedovich** [comment]
Hello @nizhikov 
Query plan contains table names and fields which are part of the schema. Schema could be treated as sensitive. Please follow [the link](https://security.stackexchange.com/questions/136045/should-company-protect-db-schema-at-the-same-level-of-the-data-itself) as proof.
Also, I know cases, when some companies (e.g. most of Banks) can't share logs thats, contains any information related to data schema.

**Comment by nizhikov** [comment]
@ygerzhedovich I think the term "sensitive" mentioned here in two contexts.

1. Type of the information that required a certain access level.
2. Information hidden when `IgniteSystemProperties#IGNITE_TO_STRING_INCLUDE_SENSITIVE = false`.

So

Is it required to restrict access to schema information? - Yes, definitely.
Is it required to hide schema information from log or any other system data - I'm not sure.

Note, access to a log file or system view should be granted only for some kind of administrator.

**Pull request closed without merging**